### ISSUE TITLE
Scatter chart with marker shape and size

### ISSUE THREAD

**Issue by wylieconlon** [elastic/elastic-charts#502]
Scatter charts and heatmaps (https://github.com/elastic/elastic-charts/issues/501) are complementary. Scatter charts use continuous scales with discrete data, while heatmaps use discrete scales and data.

Scatter charts should support:

* X and Y positioning
* Filled circle and outline markers at the least
* (optional) Markers that take a dynamic shape for categorical data
* Marker size based on value
* Tooltip labeling

<img width="635" alt="Screenshot 2019-12-19 16 17 19" src="https://user-images.githubusercontent.com/666475/71210521-12cd7d00-227b-11ea-8c20-1160ace5b749.png">
<img width="870" alt="Screenshot 2019-12-19 16 17 24" src="https://user-images.githubusercontent.com/666475/71210523-12cd7d00-227b-11ea-83a2-1d74d499db27.png">
<img width="325" alt="Screenshot 2019-12-19 16 17 30" src="https://user-images.githubusercontent.com/666475/71210524-12cd7d00-227b-11ea-8e4f-a2800ee55478.png">

**Kibana Cross Issues**
https://github.com/elastic/kibana/issues/17076
https://github.com/elastic/kibana/issues/20976
https://github.com/elastic/kibana/issues/36541

**Checklist**

- [x] this request is checked against already exist requests
- [x] every related Kibana issue is listed under `Kibana Cross Issues` list
- [x] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present

**Comment by nickofthyme** [comment]
More examples:
- https://vega.github.io/vega-lite/examples/point_2d.html
- https://vega.github.io/vega-lite/examples/tick_dot.html

**Comment by wylieconlon** [comment]
An important element of these example scatter charts is that the legend includes a shape and color indicator without showing every individual value. Supporting this type of legend seems like a nice-to-have.

**Comment by AlonaNadler** [comment]
@markov00 APM team plans to use scatterplot in 7.11 https://github.com/elastic/kibana/issues/81720
Based on their needs, do you think what Elastic charts currently has can help ?

**Comment by markov00** [comment]
@AlonaNadler we can probably take that effort to add a simple scatterplot, but it will just be a throw-away implementation as we can't have a fully-featured implementation of a real scatterplot in few weeks. We can, as we have done for the heatmap, support APM on building that directly in elastic-charts, building together a good starting point

**Comment by sqren** [comment]
@markov00 Can we avoid the throw-away work if we schedule this for 7.12, thereby giving you more time? I also recall you talking about improved elasticsearch support for this. If we extend the timeline, maybe we can involve them as well?

**Comment by markov00** [comment]
@sqren I think we can work with that time schedule.
I will prepare a POC in the next weeks or less (I'd like to check if the current configuration we have for bubble charts can work as it is for your use case.
I've one question for you: could you please write down a list of requirements for the POC and the final goal?
I'd like to have also a list of use cases, some testing datasets and edge cases (max number of points, number of different series represented, min-max values, can we go negative on Y and X?)
Thanks

**Comment by markov00** [comment]
@sqren can you please check if this can work for your case: https://codesandbox.io/s/suspicious-galileo-k3nzs?file=/src/App.tsx

It's is a scatterplot with linear axes, series colored differently, resizable point markers, tooltip and legend. It's probably also possible to add brushing but I've to check.

If you think it's fine for you, you can already use the same chart configuration

**Comment by smith** [comment]
@markov00 

> @sqren can you please check if this can work for your case: https://codesandbox.io/s/suspicious-galileo-k3nzs?file=/src/App.tsx

Yes that totally works. At this point we don't need brushing at all.

**Comment by sqren** [comment]
@markov00 That looks very promising! We'll start with that and let you know if we run into any blockers. If this works out there's nothing stopping us from releasing in 7.11, is there?

**Comment by markov00** [comment]
@sqren sure, if that works for you you can go on and release that in 7.11

**Comment by odmby** [comment]
is there any ETA for scatter plots in kibana? (except for canvas which cannot be added to dashboards)

**Comment by markov00** [comment]
> is there any ETA for scatter plots in kibana? (except for canvas which cannot be added to dashboards)

Hi @odmby we don't have an ETA yet. We will post an update when something will move in that direction
